### PR TITLE
Handle empty response exception from the kore client

### PR DESCRIPTION
**Pull request by iFrostizz**
When the response from the `JsonRpcClient` is empty (usually happens when there is a memory leak in the backend), no handling of the response is done and the proof will crash on a JsonDecodeError that may not be very meaningful for the user.
This PR is a fix for that that returns a `JsonRpcError` with the appropriate code and tests.

### CODE REVIEW

File: src/pyk/kore/rpc.py [hunk 1]
```diff
@@ -232,6 +232,8 @@ def request(self, method: str, **params: Any) -> dict[str, Any]:
 
         _LOGGER.debug(f'Sending request to {server_addr}: {req}')
         resp = self._transport.request(req)
+        if len(resp) == 0:
```
Comment (tothtamas28) [inline src/pyk/kore/rpc.py:235]:
```suggestion
        if not resp:
```

File: src/pyk/kore/rpc.py [hunk 2]
```diff
@@ -232,6 +232,8 @@ def request(self, method: str, **params: Any) -> dict[str, Any]:
...
+            raise JsonRpcError('Parse error: Empty response received', -32700)
```
Comment (tothtamas28) [inline src/pyk/kore/rpc.py:236]: This indicates that the request received by the server cannot be parsed. Instead, this should raise a `RuntimeError` or similar.

File: src/tests/unit/kore/test_client.py [hunk 3]
```diff
@@ -256,3 +258,66 @@ def test_add_module(
 
     # Then
     rpc_client.assert_request('add-module', **params)
+
+
+class MockTransport:
```
Comment (tothtamas28) [inline src/tests/unit/kore/test_client.py:263]: Consider moving these tests to a new module (e.g. `test_rpc_client.py`). Then there's no conflict with old names and things can have more concise names.

File: src/pyk/kore/rpc.py (hunk 1 above)
Comment (iFrostizz) [reply src/pyk/kore/rpc.py:235]: Thanks!

File: src/pyk/kore/rpc.py (hunk 2 above)
Comment (iFrostizz) [reply src/pyk/kore/rpc.py:236]: Makes sense, thank you!

File: src/tests/unit/kore/test_client.py (hunk 3 above)
Comment (iFrostizz) [reply src/tests/unit/kore/test_client.py]: Now moved to a new module, thanks!

File: src/pyk/kore/rpc.py [hunk 4]
```diff
@@ -232,6 +232,8 @@ def request(self, method: str, **params: Any) -> dict[str, Any]:
 
         _LOGGER.debug(f'Sending request to {server_addr}: {req}')
         resp = self._transport.request(req)
+        if not resp:
+            raise RuntimeError('Parse error: Empty response received')
```
Comment (tothtamas28) [inline src/pyk/kore/rpc.py:236]:
```suggestion
            raise RuntimeError('Empty response received')
```

**Review by tothtamas28 — APPROVED** [review]
Looks good, thank you!

**Pull request merged**